rbd.c

```diff
@@ -239,6 +249,103 @@ static void tcmu_rbd_service_status_update(struct tcmu_device *dev,
 
 #endif /* LIBRADOS_SUPPORTS_SERVICES */
 
+#if defined LIBRBD_SUPPORTS_GETADDRS || defined RBD_LOCK_ACQUIRE_SUPPORT
+static void tcmu_rbd_rm_stale_entry_from_blacklist(struct tcmu_device *dev, struct blacklist_entry *entry)
+{
+	struct tcmu_rbd_state *state = tcmur_dev_get_private(dev);
+	const char *p = entry->addrs, *q, *end;
+	char *cmd, *addr;
+	int ret;
+
+	/*
+	 * The entry->addrs will a string like:
+	 * "[192.168.195.172:0/2203456141,192.168.195.172:0/4908756432]"
+	 * Or
+	 * "192.168.195.172:0/2203456141"
+	 */
+	while (1) {
+		if (p == NULL || *p == ']') {
+			entry->is_removed = true;
+			return;
+		} else if (*p == '[' || *p == ',') {
+			/* Skip "[ " */
+			while (!isalnum(*p)) p++;
```

Comment:
Nit: avoid `NULL` as well

Reply:
The string is `'\0'` ternimated, so there is no any chance it could be `NULL` in the `else if` case.

Reply:
If you have "[\0" as a string, wouldn't this just keep looping past the bounds?

Reply:
BTW, in which case will this happen ? 

From the ceph code, it should always be '\0' terminated c-string for this address(es), it will be coverted from a vector if there have more than one addresses. And also from my local test the string coverted from a verctor will always be '\0' terminated.

Did I miss something about this ?

Reply:
Oh, maybe we should switch to use client_lock.is_locked_by_me(), right.

Reply:
No, I am just suggesting you defend against malformed data.

Reply:
Okay, will fix it. Thanks.